build/common.sh

```diff
@@ -833,3 +833,43 @@ function kube::release::gcs::publish_latest() {
   echo "+++ gsutil cat ${latest_file_dst}:"
   gsutil cat ${latest_file_dst}
 }
+
+# Publish a new latest.txt, but only if the release we're dealing with
+# is newer than the contents in GCS.
+function kube::release::gcs::publish_latest_official() {
+  local new_version=${KUBE_GCS_LATEST_CONTENTS}
+  local latest_file_dst="gs://${KUBE_GCS_RELEASE_BUCKET}/${KUBE_GCS_LATEST_FILE}"
+
+  local version_regex="^v(0|[1-9][0-9]*)\\.(0|[1-9][0-9]*)\\.(0|[1-9][0-9]*)$"
+  [[ ${new_version} =~ ${version_regex} ]] || {
+    echo "!!! publish_latest_official passed bogus value: '${new_version}'" >&2
+    exit 1
```

Comment:
`return 1` might be slightly better here, since then you can check it from the caller, and if you just call the function with no error handling, then `errexit` should cause it to just exit 1 in that case anyways...

Here and elsewhere int his function.